src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import classNames from 'classnames';
+import { useParams } from 'react-router-dom';
+import { Loader } from './Loader';
+import { Person } from '../types';
+import { fetchPeopleData } from '../utils/fetchPeopleData';
+import { generateNavLink } from '../utils/generateNavLink';
+
+export const PeoplePage = () => {
+  const [peopleData, setPeopleData] = React.useState<Person[] | null>(null);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [isError, setError] = React.useState(false);
+
+  const { personID } = useParams();
+
+  React.useEffect(() => {
+    setIsLoading(true);
+
+    fetchPeopleData()
+      .then((data) => {
+        setPeopleData(data);
+      })
+      .catch(() => {
+        setError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const findPerson = (personName: string | null) => {
```

Comment:
The 'findPerson' function is being used within the rendering of the component, which could impact performance as it runs on every render. Consider using React.useMemo to memoize the function result and improve performance.